corehq/apps/hqadmin/reports.py

```diff
@@ -266,3 +273,73 @@ def _format_date(date):
         if date:
             return parse(date).strftime(SERVER_DATETIME_FORMAT)
         return "---"
+
+
+class DeployHistoryReport(GetParamsMixin, AdminReport):
+    base_template = 'reports/base_template.html'
+
+    slug = 'deploy_history_report'
+    name = ugettext_lazy("Deploy History Report")
+
+    emailable = False
+    exportable = False
+    ajax_pagination = True
+    default_rows = 10
+
+    @property
+    def headers(self):
+        return DataTablesHeader(
+            DataTablesColumn(_("Date"), sortable=False),
+            DataTablesColumn(_("User"), sortable=False),
+            DataTablesColumn(_("Diff URL"), sortable=False),
+        )
+
+    @property
+    def rows(self):
+        deploy_list = HqDeploy.objects.all()
+        start = self.pagination.start
+        end = self.pagination.start + self.pagination.count
+        for deploy in deploy_list[start:end]:
+            yield [
+                self._format_date(deploy.date),
+                deploy.user,
+                self._hyperlink_diff_url(deploy.diff_url),
+            ]
+
+    @property
+    def total_records(self):
+        return HqDeploy.objects.count()
+
+    @cached_property
+    def _user_lookup_url(self):
+        return reverse('web_deploy_lookup')
+
+    def _format_date(self, date):
+        if date:
+            raw_time_since_deploy = dt.now() - date
+            delta_dict = self._strfdelta(raw_time_since_deploy)
+
+            if delta_dict['days'] != 0:
+                delta_str = "{days} day(s), {hours} hour(s) ago".format(
+                    **delta_dict)
+            elif delta_dict['hours'] != 0:
+                delta_str = "{hours} hour(s), {minutes} minute(s) ago".format(
+                    **delta_dict)
+            else:
+                delta_str = "{minutes} minute(s), {seconds} second(s) ago".format(
+                    **delta_dict)
+
+            ret_str = '<div>{delta}</div><div>{actual_date}</div>'
+            return ret_str.format(delta=delta_str, actual_date=date.strftime(SERVER_DATETIME_FORMAT))
+        return "---"
+
+    def _strfdelta(self, tdelta):
+        # datetime.timedelta object cannot use 'strftime', use custom fn:
+        relative_time_formatted = {"days": tdelta.days}
+        relative_time_formatted["hours"], rem = divmod(tdelta.seconds, 3600)
+        relative_time_formatted["minutes"], relative_time_formatted[
+            "seconds"] = divmod(rem, 60)
+        return relative_time_formatted
+
+    def _hyperlink_diff_url(self, diff_url):
+        return '<a href="{link}">Diff with previous</a>'.format(link=diff_url)
```

Comment:
```suggestion
        return f'<a href="{diff_url}">Diff with previous</a>'
```
Happily, HQ is on python 3, so you can use f-strings!